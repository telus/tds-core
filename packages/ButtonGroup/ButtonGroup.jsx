import React from 'react'
import PropTypes from 'prop-types'

import Box from '@tds/core-box'
import { componentWithName } from '@tds/util-prop-types'

import ButtonGroupItem from './ButtonGroupItem/ButtonGroupItem'

import safeRest from '../../shared/utils/safeRest'

import styles from './ButtonGroup.modules.scss'

/**
 * @version ./package.json
 */
const ButtonGroup = ({ name, onChange, value, children, ...rest }) => {
  const getButtons = () => {
    return React.Children.map(children, child =>
      React.cloneElement(child, { name, onChange, selectedValue: value })
    )
  }
  return (
    <fieldset>
      <Box between={3} inline dangerouslyAddClassName={styles.buttonGroup} {...safeRest(rest)}>
        {getButtons()}
      </Box>
    </fieldset>
  )
}

ButtonGroup.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  value: PropTypes.string.isRequired,
  children: componentWithName('ButtonGroupItem').isRequired,
}

ButtonGroup.defaultProps = {
  onChange: undefined,
}

ButtonGroup.Item = ButtonGroupItem

export default ButtonGroup
