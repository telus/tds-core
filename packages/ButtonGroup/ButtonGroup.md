```
initialState = {
  choice: '',
}


const onChange = (event) => {
  setState({choice: event.target.value})
}

<ButtonGroup name="storageSize" onChange={onChange} value={state.choice} label="Please select a storage size">
  <ButtonGroup.Item value="64gb">64 GB</ButtonGroup.Item>
  <ButtonGroup.Item value="128gb">128 GB</ButtonGroup.Item>
  <ButtonGroup.Item value="256gb">256 GB</ButtonGroup.Item>
</ButtonGroup>
```
