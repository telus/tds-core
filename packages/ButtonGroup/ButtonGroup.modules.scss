.buttonGroup {
  display: flex;
  flex-flow: row wrap;
  max-width: 784px;
}
