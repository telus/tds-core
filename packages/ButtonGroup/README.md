# TDS Core: ButtonGroup

An input component utilizing buttons that act as radios.
