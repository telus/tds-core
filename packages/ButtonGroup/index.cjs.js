require('./dist/index.css')
const ButtonGroup = require('./dist/index.cjs')

module.exports = ButtonGroup
