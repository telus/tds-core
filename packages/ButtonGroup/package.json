{
  "name": "@tds/core-button-group",
  "version": "2.1.16",
  "description": "An input component utilizing buttons that act as radios.",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds-core.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds-core/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "@tds/core-box": "^2.2.1",
    "@tds/core-responsive": "^1.3.3",
    "@tds/core-text": "^3.1.1",
    "@tds/shared-styles": "^1.5.2",
    "@tds/shared-typography": "^1.3.5",
    "@tds/util-helpers": "^1.4.2",
    "@tds/util-prop-types": "^1.4.0",
    "prop-types": "^15.5.10"
  },
  "devDependencies": {
    "@tds/core-colours": "^2.2.1",
    "@tds/util-generate-id": "^1.0.4"
  }
}
