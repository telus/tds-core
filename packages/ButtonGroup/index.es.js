import './dist/index.css'
import ButtonGroup from './dist/index.es'

export default ButtonGroup
