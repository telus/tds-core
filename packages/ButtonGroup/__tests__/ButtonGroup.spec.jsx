import React from 'react'
import { shallow } from 'enzyme'

import ButtonGroup from '../ButtonGroup'

describe('ButtonGroup', () => {
  const doShallow = (props = {}) => shallow(<ButtonGroup {...props} />)

  it('renders', () => {
    const buttonGroup = doShallow()

    expect(buttonGroup).toMatchSnapshot()
  })

  it('does other things', () => {
    const buttonGroup = doShallow()

    expect(buttonGroup).toExist()
  })

  it('passes additional attributes to the element', () => {
    const buttonGroup = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(buttonGroup).toHaveProp('id', 'the-id')
    expect(buttonGroup).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const buttonGroup = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(buttonGroup).not.toHaveProp('className', 'my-custom-class')
    expect(buttonGroup).not.toHaveProp('style')
  })
})
