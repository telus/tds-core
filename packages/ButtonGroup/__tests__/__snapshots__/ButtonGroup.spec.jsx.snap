// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 > *:not(:last-child) {
  margin-right: 1rem;
}

.c0 {
  color: #2a2c2e;
  word-wrap: break-word;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  word-wrap: break-word;
  font-weight: 700;
}

.c0 sup {
  top: -0.5em;
  font-size: 0.875rem;
  position: relative;
  vertical-align: baseline;
  padding-left: 0.1em;
}

.c3 {
  margin: 0.5rem 0;
}

.c4 {
  position: fixed;
  opacity: 0;
}

.c4:checked ~ label {
  background-color: #4b286d;
  box-shadow: 0px 0px 0px 0px #4b286d;
  color: #fff;
}

.c4:focus ~ label {
  box-shadow: 0px 0px 0px 2px #4b286d,0px 0px 8px 1px #4b286d;
}

.c5 {
  border-width: 0;
  border-radius: 4px;
  word-wrap: break-word;
  font-size: 1rem;
  -webkit-letter-spacing: -0.8px;
  -moz-letter-spacing: -0.8px;
  -ms-letter-spacing: -0.8px;
  letter-spacing: -0.8px;
  line-height: 1.5;
  word-wrap: break-word;
  font-weight: 700;
  font-family: "TELUS-Web","Helvetica Neue",Helvetica,Arial,sans-serif;
  -webkit-transition: background 0.2s,color 0.2s,box-shadow 0.2s;
  transition: background 0.2s,color 0.2s,box-shadow 0.2s;
  background-color: #fff;
  box-shadow: 0px 0px 0px 1px #4b286d;
  color: #4b286d;
  white-space: nowrap;
  min-width: 136px;
}

.c5:hover {
  background-color: #fff;
  color: #4b286d;
  box-shadow: 0px 0px 0px 2px #4b286d,0px 0px 8px 1px #4b286d;
}

.c1 {
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  max-width: 784px;
}

@media (min-width:768px) {
  .c5 {
    min-width: 136px;
  }
}

<ButtonGroup
  label="Please select a storage size"
  name="storageSize"
  onChange={[Function]}
  value="64gb"
>
  <fieldset
    name="storageSize"
  >
    <legend>
      <Text
        block={false}
        bold={true}
        invert={false}
        size="medium"
      >
        <Text__StyledText
          block={false}
          bold={true}
          invert={false}
          size="medium"
        >
          <StyledComponent
            block={false}
            bold={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Text__StyledText-sc-1m0rr67-0",
                  "isStatic": false,
                  "lastClassName": "c0",
                  "rules": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    "sup {",
                    "top: -0.5em;",
                    "font-size: 0.875rem;",
                    "position: relative;",
                    "vertical-align: baseline;",
                    "padding-left: 0.1em;",
                    "}",
                  ],
                },
                "displayName": "Text__StyledText",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Text__StyledText-sc-1m0rr67-0",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            invert={false}
            size="medium"
          >
            <span
              className="c0"
              size="medium"
            >
              Please select a storage size
            </span>
          </StyledComponent>
        </Text__StyledText>
      </Text>
    </legend>
    <Styled(Box)
      between={3}
      inline={true}
      tag="div"
    >
      <StyledComponent
        between={3}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "lastClassName": "c1",
              "rules": Array [
                "flex-flow: row wrap;",
                "max-width: 784px;",
              ],
            },
            "defaultProps": Object {
              "below": undefined,
              "between": undefined,
              "className": undefined,
              "horizontal": undefined,
              "inline": false,
              "inset": undefined,
              "tag": "div",
              "vertical": undefined,
            },
            "displayName": "Styled(Box)",
            "foldedComponentIds": Array [],
            "propTypes": Object {
              "below": [Function],
              "between": [Function],
              "children": [Function],
              "className": [Function],
              "horizontal": [Function],
              "inline": [Function],
              "inset": [Function],
              "tag": [Function],
              "vertical": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "below": undefined,
                "between": undefined,
                "className": undefined,
                "horizontal": undefined,
                "inline": false,
                "inset": undefined,
                "tag": "div",
                "vertical": undefined,
              },
              "displayName": "Box",
              "propTypes": Object {
                "below": [Function],
                "between": [Function],
                "children": [Function],
                "className": [Function],
                "horizontal": [Function],
                "inline": [Function],
                "inset": [Function],
                "tag": [Function],
                "vertical": [Function],
              },
              "render": [Function],
            },
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        inline={true}
        tag="div"
      >
        <Box
          between={3}
          className="c1"
          inline={true}
          tag="div"
        >
          <styled.div
            between={3}
            className="c1"
            inline={true}
            tag="div"
          >
            <StyledComponent
              between={3}
              className="c1"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [
                    [Function],
                  ],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "c2",
                    "rules": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              inline={true}
              tag="div"
            >
              <div
                className="c1 sc-bwzfXH c1 c2"
              >
                <ButtonGroup.Item
                  checked={true}
                  key=".0"
                  name="storageSize"
                  onChange={[Function]}
                  value="64gb"
                >
                  <ButtonGroupItem__StyledButtonGroupItem>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "ButtonGroupItem__StyledButtonGroupItem-drabnz-0",
                            "isStatic": true,
                            "lastClassName": "c3",
                            "rules": Array [
                              "margin: 0.5rem 0;",
                            ],
                          },
                          "displayName": "ButtonGroupItem__StyledButtonGroupItem",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "ButtonGroupItem__StyledButtonGroupItem-drabnz-0",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c3"
                      >
                        <ButtonGroupItem__StyledInput
                          checked={true}
                          id="storagesize_64gb"
                          name="storageSize"
                          onChange={[Function]}
                          type="radio"
                          value="64gb"
                        >
                          <StyledComponent
                            checked={true}
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "ButtonGroupItem__StyledInput-drabnz-1",
                                  "isStatic": true,
                                  "lastClassName": "c4",
                                  "rules": Array [
                                    "position: fixed;",
                                    "opacity: 0;",
                                    "&:checked ~ label {",
                                    "background-color: #4b286d;",
                                    "box-shadow: 0px 0px 0px 0px #4b286d;",
                                    "color: #fff;",
                                    "}",
                                    "&:focus ~ label {",
                                    "box-shadow: 0px 0px 0px 2px #4b286d, 0px 0px 8px 1px #4b286d;",
                                    "}",
                                  ],
                                },
                                "displayName": "ButtonGroupItem__StyledInput",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "ButtonGroupItem__StyledInput-drabnz-1",
                                "target": "input",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            id="storagesize_64gb"
                            name="storageSize"
                            onChange={[Function]}
                            type="radio"
                            value="64gb"
                          >
                            <input
                              checked={true}
                              className="c4"
                              id="storagesize_64gb"
                              name="storageSize"
                              onChange={[Function]}
                              type="radio"
                              value="64gb"
                            />
                          </StyledComponent>
                        </ButtonGroupItem__StyledInput>
                        <ButtonGroupItem__StyledLabel
                          htmlFor="storagesize_64gb"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "ButtonGroupItem__StyledLabel-drabnz-2",
                                  "isStatic": true,
                                  "lastClassName": "c5",
                                  "rules": Array [
                                    "border-width: 0;",
                                    "border-radius: 4px;",
                                    "word-wrap: break-word;",
                                    "font-size: 1rem;",
                                    "letter-spacing: -0.8px;",
                                    "line-height: 1.5;",
                                    "word-wrap: break-word;",
                                    "font-weight: 700;",
                                    "font-family: \\"TELUS-Web\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif;",
                                    "transition: background 0.2s, color 0.2s, box-shadow 0.2s;",
                                    "background-color: #fff;",
                                    "box-shadow: 0px 0px 0px 1px #4b286d;",
                                    "color: #4b286d;",
                                    "white-space: nowrap;",
                                    "min-width: 136px;",
                                    "&:hover {",
                                    "background-color: #fff;",
                                    "color: #4b286d;",
                                    "box-shadow: 0px 0px 0px 2px #4b286d, 0px 0px 8px 1px #4b286d;",
                                    "}",
                                    "@media (min-width: 768px) {",
                                    "min-width: 136px;",
                                    "}",
                                  ],
                                },
                                "displayName": "ButtonGroupItem__StyledLabel",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "ButtonGroupItem__StyledLabel-drabnz-2",
                                "target": "label",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            htmlFor="storagesize_64gb"
                          >
                            <label
                              className="c5"
                              htmlFor="storagesize_64gb"
                            >
                              64 GB
                            </label>
                          </StyledComponent>
                        </ButtonGroupItem__StyledLabel>
                      </div>
                    </StyledComponent>
                  </ButtonGroupItem__StyledButtonGroupItem>
                </ButtonGroup.Item>
                <ButtonGroup.Item
                  checked={false}
                  key=".1"
                  name="storageSize"
                  onChange={[Function]}
                  value="128gb"
                >
                  <ButtonGroupItem__StyledButtonGroupItem>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "ButtonGroupItem__StyledButtonGroupItem-drabnz-0",
                            "isStatic": true,
                            "lastClassName": "c3",
                            "rules": Array [
                              "margin: 0.5rem 0;",
                            ],
                          },
                          "displayName": "ButtonGroupItem__StyledButtonGroupItem",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "ButtonGroupItem__StyledButtonGroupItem-drabnz-0",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c3"
                      >
                        <ButtonGroupItem__StyledInput
                          checked={false}
                          id="storagesize_128gb"
                          name="storageSize"
                          onChange={[Function]}
                          type="radio"
                          value="128gb"
                        >
                          <StyledComponent
                            checked={false}
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "ButtonGroupItem__StyledInput-drabnz-1",
                                  "isStatic": true,
                                  "lastClassName": "c4",
                                  "rules": Array [
                                    "position: fixed;",
                                    "opacity: 0;",
                                    "&:checked ~ label {",
                                    "background-color: #4b286d;",
                                    "box-shadow: 0px 0px 0px 0px #4b286d;",
                                    "color: #fff;",
                                    "}",
                                    "&:focus ~ label {",
                                    "box-shadow: 0px 0px 0px 2px #4b286d, 0px 0px 8px 1px #4b286d;",
                                    "}",
                                  ],
                                },
                                "displayName": "ButtonGroupItem__StyledInput",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "ButtonGroupItem__StyledInput-drabnz-1",
                                "target": "input",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            id="storagesize_128gb"
                            name="storageSize"
                            onChange={[Function]}
                            type="radio"
                            value="128gb"
                          >
                            <input
                              checked={false}
                              className="c4"
                              id="storagesize_128gb"
                              name="storageSize"
                              onChange={[Function]}
                              type="radio"
                              value="128gb"
                            />
                          </StyledComponent>
                        </ButtonGroupItem__StyledInput>
                        <ButtonGroupItem__StyledLabel
                          htmlFor="storagesize_128gb"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "ButtonGroupItem__StyledLabel-drabnz-2",
                                  "isStatic": true,
                                  "lastClassName": "c5",
                                  "rules": Array [
                                    "border-width: 0;",
                                    "border-radius: 4px;",
                                    "word-wrap: break-word;",
                                    "font-size: 1rem;",
                                    "letter-spacing: -0.8px;",
                                    "line-height: 1.5;",
                                    "word-wrap: break-word;",
                                    "font-weight: 700;",
                                    "font-family: \\"TELUS-Web\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif;",
                                    "transition: background 0.2s, color 0.2s, box-shadow 0.2s;",
                                    "background-color: #fff;",
                                    "box-shadow: 0px 0px 0px 1px #4b286d;",
                                    "color: #4b286d;",
                                    "white-space: nowrap;",
                                    "min-width: 136px;",
                                    "&:hover {",
                                    "background-color: #fff;",
                                    "color: #4b286d;",
                                    "box-shadow: 0px 0px 0px 2px #4b286d, 0px 0px 8px 1px #4b286d;",
                                    "}",
                                    "@media (min-width: 768px) {",
                                    "min-width: 136px;",
                                    "}",
                                  ],
                                },
                                "displayName": "ButtonGroupItem__StyledLabel",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "ButtonGroupItem__StyledLabel-drabnz-2",
                                "target": "label",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            htmlFor="storagesize_128gb"
                          >
                            <label
                              className="c5"
                              htmlFor="storagesize_128gb"
                            >
                              128 GB
                            </label>
                          </StyledComponent>
                        </ButtonGroupItem__StyledLabel>
                      </div>
                    </StyledComponent>
                  </ButtonGroupItem__StyledButtonGroupItem>
                </ButtonGroup.Item>
                <ButtonGroup.Item
                  checked={false}
                  key=".2"
                  name="storageSize"
                  onChange={[Function]}
                  value="256gb"
                >
                  <ButtonGroupItem__StyledButtonGroupItem>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "ButtonGroupItem__StyledButtonGroupItem-drabnz-0",
                            "isStatic": true,
                            "lastClassName": "c3",
                            "rules": Array [
                              "margin: 0.5rem 0;",
                            ],
                          },
                          "displayName": "ButtonGroupItem__StyledButtonGroupItem",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "ButtonGroupItem__StyledButtonGroupItem-drabnz-0",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c3"
                      >
                        <ButtonGroupItem__StyledInput
                          checked={false}
                          id="storagesize_256gb"
                          name="storageSize"
                          onChange={[Function]}
                          type="radio"
                          value="256gb"
                        >
                          <StyledComponent
                            checked={false}
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "ButtonGroupItem__StyledInput-drabnz-1",
                                  "isStatic": true,
                                  "lastClassName": "c4",
                                  "rules": Array [
                                    "position: fixed;",
                                    "opacity: 0;",
                                    "&:checked ~ label {",
                                    "background-color: #4b286d;",
                                    "box-shadow: 0px 0px 0px 0px #4b286d;",
                                    "color: #fff;",
                                    "}",
                                    "&:focus ~ label {",
                                    "box-shadow: 0px 0px 0px 2px #4b286d, 0px 0px 8px 1px #4b286d;",
                                    "}",
                                  ],
                                },
                                "displayName": "ButtonGroupItem__StyledInput",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "ButtonGroupItem__StyledInput-drabnz-1",
                                "target": "input",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            id="storagesize_256gb"
                            name="storageSize"
                            onChange={[Function]}
                            type="radio"
                            value="256gb"
                          >
                            <input
                              checked={false}
                              className="c4"
                              id="storagesize_256gb"
                              name="storageSize"
                              onChange={[Function]}
                              type="radio"
                              value="256gb"
                            />
                          </StyledComponent>
                        </ButtonGroupItem__StyledInput>
                        <ButtonGroupItem__StyledLabel
                          htmlFor="storagesize_256gb"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "ButtonGroupItem__StyledLabel-drabnz-2",
                                  "isStatic": true,
                                  "lastClassName": "c5",
                                  "rules": Array [
                                    "border-width: 0;",
                                    "border-radius: 4px;",
                                    "word-wrap: break-word;",
                                    "font-size: 1rem;",
                                    "letter-spacing: -0.8px;",
                                    "line-height: 1.5;",
                                    "word-wrap: break-word;",
                                    "font-weight: 700;",
                                    "font-family: \\"TELUS-Web\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif;",
                                    "transition: background 0.2s, color 0.2s, box-shadow 0.2s;",
                                    "background-color: #fff;",
                                    "box-shadow: 0px 0px 0px 1px #4b286d;",
                                    "color: #4b286d;",
                                    "white-space: nowrap;",
                                    "min-width: 136px;",
                                    "&:hover {",
                                    "background-color: #fff;",
                                    "color: #4b286d;",
                                    "box-shadow: 0px 0px 0px 2px #4b286d, 0px 0px 8px 1px #4b286d;",
                                    "}",
                                    "@media (min-width: 768px) {",
                                    "min-width: 136px;",
                                    "}",
                                  ],
                                },
                                "displayName": "ButtonGroupItem__StyledLabel",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "ButtonGroupItem__StyledLabel-drabnz-2",
                                "target": "label",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            htmlFor="storagesize_256gb"
                          >
                            <label
                              className="c5"
                              htmlFor="storagesize_256gb"
                            >
                              256 GB
                            </label>
                          </StyledComponent>
                        </ButtonGroupItem__StyledLabel>
                      </div>
                    </StyledComponent>
                  </ButtonGroupItem__StyledButtonGroupItem>
                </ButtonGroup.Item>
              </div>
            </StyledComponent>
          </styled.div>
        </Box>
      </StyledComponent>
    </Styled(Box)>
  </fieldset>
</ButtonGroup>
`;
